modules/pool/lib/pool-on-chain-data.service.ts

```diff
@@ -402,6 +408,33 @@ export class PoolOnChainDataService {
                     });
                 }
 
+                // Save GyroEV2 tokenRates taken from multicall result  
+                if (isGyroEV2(pool)) {
+                    const tokenRates =
+                        typeof multicallResult?.tokenRates !== 'undefined'
+                            ? multicallResult.tokenRates.map((r) => r.toString())
+                            : pool.gyroDynamicData?.tokenRates;
+                    if (
+                        !pool.gyroDynamicData ||
+                        !(
+                            tokenRates &&
+                            pool.gyroDynamicData.tokenRates.every((element, i) => tokenRates[i] === element)
+                        )
+                    ) {
+                        await prisma.prismaPoolGyroDynamicData.upsert({
+                            where: { id_chain: { id: pool.id, chain: networkContext.chain } },
+                            create: {
+                                id: pool.id,
+                                chain: networkContext.chain,
+                                poolId: pool.id,
+                                tokenRates,
+                                blockNumber,
+                            },
+                            update: { tokenRates, blockNumber },
+                        });
+                    }
+                }
+
                 for (const poolToken of pool.tokens) {
```

Comment:
You should add the rate under the poolToken instead of creating a new dataset for this. Similar to what we do for the linear token rates.